lib/dradis/plugins/template_service.rb

```diff
@@ -1,29 +1,22 @@
 module Dradis
   module Plugins
     class TemplateService
-      attr_accessor :logger, :template, :templates_dir
+      attr_accessor :logger, :plugin, :processor, :project, :template, :templates_dir
 
       def initialize(args = {})
         @plugin        = args.fetch(:plugin)
         @templates_dir = args[:templates_dir] || default_templates_dir
+        @project       = args.fetch(:project, nil)
       end
 
       # For a given entry, return a text blob resulting from applying the
-      # chosen template to the supplied entry.
+      # appropriate mapping to the supplied entry.
       def process_template(args = {})
         self.template = args[:template]
         data          = args[:data]
+        @processor    = plugin::FieldProcessor.new(data: data)
 
-        processor = @plugin::FieldProcessor.new(data: data)
-
-        template_source.gsub(/%(\S*?)%/) do |field|
-          name = field[1..-2]
-          if fields.include?(name)
-            processor.value(field: name)
-          else
-            "Field [#{field}] not recognized by the plugin"
-          end
-        end
+        apply_mapping(mapping)&.join("\n\n")
```

Comment:
At a high enough level `apply_mapping` is functional programming. It also seems like it's the kind of operation that we're going to be using again and again: in the UI, in the exporters, etc.

Wouldn't something like `mapping#apply()` make more sense?

I realise #process_tempalte is pretty functional too. So perhaps not something to address in this PR, but something to keep in the back of your mind for the cycle. There's a whiff to it.

Reply:
> Wouldn't something like mapping#apply() make more sense?

Do you mean in the main app? An `apply` message sent to a mapping instance?